janitor/utils.py

```diff
@@ -405,3 +406,219 @@ def _wrapped(x, *args, **kwargs):
+    This checks if entry is a dictionary,
+    checks the instance of value in key:value pairs in entry,
+    and makes changes to other types as deemed necessary. 
+    """
+    # entry has to be a dictionary
+    if not entry:
+        raise ValueError("passed dictionary cannot be empty")
+    # if it is a number or string, wrap in a list
+    entry = {
+        key: [value]
+        if isinstance(value, (type(None), int, float, bool, str))
+        else value
+        for key, value in entry.items()
+    }
+
+    # convert to list if value is a string|tuple|range
+    entry = {
+        key: list(value) if isinstance(value, (set, tuple, range)) else value
+        for key, value in entry.items()
+    }
+
+    # collect dataframes here
+    dfs = []
+
+    # collect non dataframes here, proper dicts ... key value pair where the value is a list of scalars
+    dicts = {}
+
+    for key, value in entry.items():
+
+        # exclude dicts:
+        if isinstance(value, dict):
+            raise ValueError("nested dicts not allowed")
+
+        # process arrays
+        if isinstance(value, np.ndarray):
+            if value.size == 0:
+                raise ValueError("array cannot be empty")
+            elif value.ndim == 1:
+                dfs.append(pd.DataFrame(value, columns=[key]))
+            elif value.ndim == 2:
+                dfs.append(pd.DataFrame(value).add_prefix(f"{key}_"))
+            else:
+                raise ValueError(
+                    "expand_grid works with only vector and matrix arrays"
+                )
+        # process series
+        if isinstance(value, pd.Series):
+            if value.empty:
+                raise ValueError("passed Series cannot be empty")
+            if not isinstance(value.index, pd.MultiIndex):
+                if value.name:
+                    value = value.to_frame(name=f"{key}_{value.name}")
+                    dfs.append(value)
+                else:
+                    value = value.to_frame(name=f"{key}")
+                    dfs.append(value)
+            else:
+                raise ValueError(
+                    "expand_grid does not work with pd.MultiIndex"
+                )
+        # process dataframe
+        if isinstance(value, pd.DataFrame):
+            if value.empty:
+                raise ValueError("passed DataFrame cannot be empty")
+            if not (
+                isinstance(value.index, pd.MultiIndex)
+                or isinstance(value.columns, pd.MultiIndex)
+            ):
+                # add key to dataframe columns
+                value = value.add_prefix(f"{key}_")
+                dfs.append(value)
+            else:
+                raise ValueError(
+                    "expand grid does not work with pd.MultiIndex"
+                )
+        # process lists
+        if isinstance(value, list):
+            if not value:
+                raise ValueError("passed value cannot be empty")
```

Comment:
The error message here is a bit vague. Any chance it could be made a bit more specific, in the same mould as the other ones I've pointed out?